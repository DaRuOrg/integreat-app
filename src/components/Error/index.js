import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'
import FontAwesome from 'react-fontawesome'

import { history } from 'main'

import style from './style.css'

class Error extends React.Component {
  static propTypes = {
    error: PropTypes.string.isRequired
  }

  /**
   * Go back in history!!
   * @param event The click event
   */
  goBack = (event) => {
    event.preventDefault()
    history.goBack()
  }

  render () {
    const {t} = this.props
    return <div>
      <div className={style.centerText}>{ t(this.props.error) }</div>
      <div className={style.centerText}><FontAwesome name='frown-o' size="5x"/></div>
      <a className={style.centerText} href="/" onClick={this.goBack}>Go back</a>
    </div>
  }
}

export default translate('errors')(Error)
