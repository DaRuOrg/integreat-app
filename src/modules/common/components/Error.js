import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'
import FontAwesome from 'react-fontawesome'

import style from './Error.css'

export class Error extends React.Component {
  static propTypes = {
    error: PropTypes.string.isRequired
  }

  render () {
    return <div>
      <div className={style.centerText}>{this.props.t(this.props.error)}</div>
      <div className={style.centerText}><FontAwesome name='frown-o' size='5x' /></div>
      <a className={style.centerText} href='/' onClick={this.goBack}>Go back</a>
    </div>
  }
}

export default translate('errors')(Error)
