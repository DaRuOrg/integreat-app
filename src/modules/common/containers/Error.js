import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'
import FontAwesome from 'react-fontawesome'
import { goBack } from 'redux-little-router'
import { connect } from 'react-redux'
import compose from 'lodash/fp/compose'

import style from './Error.css'

export class Error extends React.Component {
  static propTypes = {
    error: PropTypes.string.isRequired,
    goBack: PropTypes.func.isRequired
  }

  /**
   * Go back in history!!
   * @param event The click event
   */
  goBack (event) {
    event.preventDefault()
    this.props.goBack()
  }

  render () {
    return <div>
      <div className={style.centerText}>{this.props.t(this.props.error)}</div>
      <div className={style.centerText}><FontAwesome name='frown-o' size='5x' /></div>
      <a className={style.centerText} href='/' onClick={this.goBack}>Go back</a>
    </div>
  }
}

const mapDispatchToProps = (dispatch) => ({
  goBack: () => dispatch(goBack())
})

export default compose(
  connect(() => {}, mapDispatchToProps),
  translate('errors')
)(Error)
