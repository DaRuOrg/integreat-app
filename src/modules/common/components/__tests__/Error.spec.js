import React from 'react'
import { shallow } from 'enzyme'

import { Error } from '../Error'

jest.mock('react-i18next')

describe('Error', () => {
  const mockTranslate = jest.fn((msg) => msg)

  test('should match snapshot', () => {
    const wrapper = shallow(
      <Error error='Error Message' t={mockTranslate} />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
