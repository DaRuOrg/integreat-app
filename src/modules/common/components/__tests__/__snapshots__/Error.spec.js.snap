// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match snapshot 1`] = `
<div>
  <div
    className="centerText"
  >
    Error Message
  </div>
  <div
    className="centerText"
  >
    <FontAwesome
      name="frown-o"
      size="5x"
    />
  </div>
  <a
    className="centerText"
    href="/"
  >
    Go back
  </a>
</div>
`;
